### PR TITLE
Clean out user from the task store's

### PR DESCRIPTION
**Pull request by lyondhill**
completes step 4 of #11427

### CODE REVIEW

File: task/platform_adapter.go [hunk 1]
```diff
@@ -12,6 +12,10 @@ import (
 	"github.com/influxdata/influxdb/task/options"
 )
 
+// TODO(lh): A temporary error response for all finds that use user id's
+// This should be removed when we transition to the new kv store.
+var ErrTempDisabled = errors.New("temporarily disabled")
```
Comment (mark-rushakoff) [inline task/platform_adapter.go]: We started but didn't resolve an offline conversation about this.

We shouldn't be using this error. The platform adapter should instead accept a UserResourceMappingService, and handle the User filter case by looking up that user's tasks and then calling `FindTaskByID` on each individual task.

We will refactor that to a more efficient approach as part of the proposed refactor in https://github.com/influxdata/influxdb/issues/11063.

Comment (lyondhill) [reply task/platform_adapter.go]: I will make that change. It should be a clean change and allow it to be a little more permanent then the current solution.

File: task/platform_adapter.go [hunk 2]
```diff
@@ -161,8 +206,28 @@ func (p pAdapter) UpdateTask(ctx context.Context, id platform.ID, upd platform.T
...
+
+	// clean up resource maps for deleted task
+	urms, _, err := p.urm.FindUserResourceMappings(ctx, platform.UserResourceMappingFilter{
+		ResourceID:   id,
+		ResourceType: platform.TasksResourceType,
+	})
+
+	if err != nil {
+		return err
+	} else {
```
Comment (mark-rushakoff) [inline task/platform_adapter.go]: Nit: could remove the `else` here so that the for loop isn't indented.

File: task/servicetest/servicetest.go [hunk 3]
```diff
@@ -174,17 +174,17 @@ func testTaskCRUD(t *testing.T, sys *System) {
...
 	}
-	f = findTask(fs, task.ID)
+	f = findTask(fs, tsk.ID)
 	found["FindTasks with Organization filter"] = f
 
 	fs, _, err = sys.ts.FindTasks(sys.Ctx, platform.TaskFilter{User: &cr.UserID})
 	if err != nil {
 		t.Fatal(err)
+	} else {
+		f = findTask(fs, tsk.ID)
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: This could/should be moved out of the `else` too. `else` looks unusual after a `t.Fatal`.

**Pull request merged**